x-pack/plugin/frozen-indices/src/test/java/org/elasticsearch/index/engine/FrozenIndexTests.java

```diff
@@ -122,26 +123,48 @@ public void testSearchAndGetAPIsAreThrottled() throws InterruptedException, IOEx
         int numRefreshes = 0;
         for (int i = 0; i < numRequests; i++) {
             numRefreshes++;
-            switch (randomIntBetween(0, 3)) {
+            // make sure that we don't share the frozen reader in concurrent requests since we acquire the
+            // searcher and rewrite the request outside of the search-throttle thread pool
+            CountDownLatch reqLatch = new CountDownLatch(1);
```

Comment:
Since we no longer execute requests concurrently, we can drop both latches and use a blocking `get()` instead of non-blocking `execute().

Reply:
Indeed, thanks !

Reply:
https://github.com/elastic/elasticsearch/pull/51708/commits/662972cf2de9b2ad36077b886c4930c8d2a08468